### DISCUSSION TITLE
Usage of firebase function & ssr mode

### DISCUSSION THREAD

**Discussion by brrinta** [quasarframework/quasar#9411] [CLI - SSR mode]
How can deploy quasar v2 ssr application into firebase-functions?

In v1, it can be done by following step:

First, I insert code at bottom in src-ssr/index.js file:

`const functions=require("firebase-functions"); exports.samplefun=functions.https.onRequest(app);`

But in new version, there are no file "index.js" in src-ssr directory.

Please help me to resolve the problem.

**Comment by l1qu1d** [comment]
I'm also running into issues when trying to deploy ssr express to firebase. I suspect it has something to do with the new ```ssr_middlewares``` implementation as when doing a local deploy it pseudo works via port 3000 but with firebase (google cloud functions) you can't use app.listen in an express server at all. In v1 you use to be able to just comment out the app.listen part, but that doesn't work anymore.

Not only that, the output ssr index.js is now compressed so it makes debugging harder. I'm sure there's a way to opt out of compressing that file, but I'm not sure how.

**Reply by l1qu1d** [reply]
Made a solution to the problem. Here's the working repo with a full explanation.
https://github.com/l1qu1d/quasar-v2-ssr-firebase

**Reply by Seanitzel** [reply]
This is awesome, your solution really works, thank you so much!

However I believe there should be a better solution for this - it doesn't make sense that we would have to manually edit an autogenerated file everytime we want to make a release lol

**Reply by Seanitzel** [reply]
This was fixed, Thanks Quasar team! <3

**Comment by hawkeye64** [comment]
@rstoenescu Does your recent commit also resolve this one? https://github.com/quasarframework/quasar/commit/5f813ee3145e3afaab4e46a3430e0e87088d795e

**Comment by vlombardifiters** [comment]
Issue confirmed my side.
@quasar/app: 3.3.3
quasar: 2.5.5

Before with Quasar v2 SSR :
```javascript
 const functions = require("firebase-functions");
 const runtimeOpts = {
    timeoutSeconds: 540,
    memory: "1GB"
  };
  exports.mywebsite = functions
    .region("us-central1")
    .runWith(runtimeOpts)
    .https.onRequest(app);
```

Now with Quasar v3 SSR :
```javascript
import * as functions from "firebase-functions";
import { ssrProductionExport } from "quasar/wrappers";

// Define API
const runtimeOpts = {
  timeoutSeconds: 540,
  memory: "1GB",
};

export default ssrProductionExport(({ ssrHandler }) => {
  return {
    mywebsite: functions
      .region("us-central1")
      .runWith(runtimeOpts)
      .https.onRequest(ssrHandler),
  };
});
```

Any advise?

**Comment by Johfr** [comment]
Hi, i'm looking for a real method to use SSR with firebase too.

Stack: Quasar CLI with Vue

**Comment by Nakroma** [comment]
For me the Firebase example in [the docs](https://quasar.dev/quasar-cli-vite/developing-ssr/ssr-webserver#example-firebase-function) didn't work at all for me, I had to turn `listen` in a synchronous function and then change my function rewrite to `default-<my_handler_name>`. Additionally I had to set the `engines.node` field in the package.json to a strict 18 (16, 14, etc. would probably also work):
```js
extendPackageJson (pkg) {
        pkg.engines.node = '18'
}
```
